src/perf/lib/Tcp.h

```diff
@@ -135,23 +135,33 @@ class TcpWorker {
     );
 };
 
-class TcpServer {
-    friend class TcpEngine;
-    bool Initialized;
-    TcpEngine* Engine;
-    CXPLAT_SEC_CONFIG* SecConfig;
-    CXPLAT_SOCKET* Listener;
+class TcpSecConfig {
+    CXPLAT_EVENT CallbackEvent;
     static
     _IRQL_requires_max_(PASSIVE_LEVEL)
     _Function_class_(CXPLAT_SEC_CONFIG_CREATE_COMPLETE)
     void
     QUIC_API
     SecConfigCallback(
-        _In_ const QUIC_CREDENTIAL_CONFIG* CredConfig,
+        _In_ const QUIC_CREDENTIAL_CONFIG* /* CredConfig */,
         _In_opt_ void* Context,
         _In_ QUIC_STATUS Status,
         _In_opt_ CXPLAT_SEC_CONFIG* SecurityConfig
         );
+public:
+    CXPLAT_SEC_CONFIG* SecConfig{nullptr};
+    TcpSecConfig() noexcept;
+    TcpSecConfig(const QUIC_CREDENTIAL_CONFIG* CredConfig) noexcept;
+    ~TcpSecConfig() noexcept;
```

Comment:
Please delete or implement the copy and move constructors if you add a destructor (rule of 5)

Reply:
We don't do that for anything else, so I prefer to not do it here, especially since it's never going to be used for this one.

Reply:
Well, not doing it on anything else is a problem too, but I can't really ask you to fix them all at once...
If the copy/move are not going to be used, deleting them enforces it. It can avoid hard to debug double free / memory corruptions letter for the price of adding 4 lines right now.

```
TcpSecConfig(const TcpSecConfig&) = delete;
TcpSecConfig(TcpSecConfig&&) = delete;
TcpSecConfig& operator=(const TcpSecConfig&) = delete;
TcpSecConfig& operator=(TcpSecConfig&&) = delete;
```

Reply:
How about this: I will do a follow up PR to delete all these for the TCP C++ classes immediately after this is merged?